src/injector/src/main.rs

```diff
@@ -154,23 +222,39 @@ fn start_seed_registry() {
 }
 
 /// Handles the GET request for the manifest (only returns a OCI manifest regardless of Accept header)
-fn handle_get_manifest(root: &Path) -> Response {
+fn handle_get_manifest(root: &Path, name: &String, tag: &String) -> Response {
     let index = fs::read_to_string(root.join("index.json")).expect("read index.json");
     let json: Value = serde_json::from_str(&index).expect("unable to parse index.json");
-    let sha_manifest = json["manifests"][0]["digest"]
-        .as_str()
-        .unwrap()
-        .strip_prefix("sha256:")
-        .unwrap()
-        .to_owned();
-    let file = File::open(&root.join("blobs").join("sha256").join(&sha_manifest)).unwrap();
-    Response::from_file("application/vnd.docker.distribution.manifest.v2+json", file)
-        .with_additional_header(
-            "Docker-Content-Digest",
-            format!("sha256:{}", sha_manifest.to_owned()),
-        )
-        .with_additional_header("Etag", format!("sha256:{}", sha_manifest))
-        .with_additional_header("Docker-Distribution-Api-Version", "registry/2.0")
+    let mut sha_manifest = "".to_owned();
+
+    if tag.starts_with("sha256:") {
+        sha_manifest = tag.strip_prefix("sha256:").unwrap().to_owned();
+    } else {
+        for manifest in json["manifests"].as_array().unwrap() { 
+            let image_base_name = manifest["annotations"]["org.opencontainers.image.base.name"].as_str().unwrap();
+            let requested_reference = name.to_owned() + ":" + tag;
+            if requested_reference == image_base_name {
+                sha_manifest = manifest["digest"].as_str()
+                    .unwrap()
+                    .strip_prefix("sha256:")
+                    .unwrap()
+                    .to_owned();
+            }
+        }
+    }
+
+    if sha_manifest != "" {
+        let file = File::open(&root.join("blobs").join("sha256").join(&sha_manifest)).unwrap();
+        Response::from_file("application/vnd.docker.distribution.manifest.v2+json", file)
+            .with_additional_header(
+                "Docker-Content-Digest",
```

Comment:
string headers like this that are common / the same across requests should prob be top level constants